src/alchemlyb/estimators/mbar_.py

```diff
@@ -1,7 +1,12 @@
+from __future__ import annotations
```

Comment:
Which versions of python support annotations?

Reply:
3.10 supports full functionality of annotations. The old versions only supports less optimal way of annotating it.

Reply:
This is an issue if we don't have `from __future__ import annotations`. Adding this allow py3.9 to be supported.